deploy.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+if (( ${#} < 3)); then
+  echo "USAGE: ${0} s3_package_bucket_name aws_region recipient_email"
+  exit 1
+fi
+
+###
+# Variables
+###
+export SAM_TEMPLATE="template.yml"
+export PACKAGED_TEMPLATE="packaged-template.yml"
+export STACK_NAME="lambda-file-refarch"
+export S3_CODE_BUCKET=${1}
+export AWS_REGION=${2}
+export RECIPIENT_EMAIL=${3}
+export SAM_BUILD_ARG=${4} # Possible to pass in --skip-pull-image if docker image is already local
+export FILE_TO_UPDATE="src/conversion/conversion.py"
+
+###
+# Functions
+###
+function sam-runner {
+    sam build --use-container ${SAM_BUILD_ARG}
+    sam package --output-template-file ${PACKAGED_TEMPLATE} \
+        --s3-bucket ${S3_CODE_BUCKET} # 2> /dev/null
+    sam deploy \
+        --template-file ${PACKAGED_TEMPLATE} \
+        --stack-name ${STACK_NAME} \
+        --region ${AWS_REGION} \
+        --tags Project=lambda-refarch-fileprocessing \
+        --parameter-overrides AlarmRecipientEmailAddress=${RECIPIENT_EMAIL} \
+        --capabilities CAPABILITY_IAM
+}
+
+###
+# Main body
```

Comment:
I appreciate what you are doing here but I think it's the wrong approach, this will break really easily and it's not really usable in a pipeline. I think the input bucket is unique enough as it contains the account ID. I am not a fan of static resource names (Due to potential update issues) but in this scenario I think it is justified, and the common issues with this don't really apply to an S3 Bucket.  I have made a comment at the code that needs changing to suggest the alternative.

Reply:
I'm going to remove deploy.sh entirely.  I'll leave the `cleanup.sh` script in place though.